src/utils/SafeString.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * SafeString is a utility function that converts a value to a string.
+ * It handles the problematic case of plain objects by converting them to JSON.
+ * It helps with eslint rule https://typescript-eslint.io/rules/no-base-to-string
+ * @param value - The value to convert to a string.
+ * @returns The string representation of the value.
+ */
+
+export default function SafeString(value: unknown): string {
+    if (value === undefined || value === null) {
+        return '';
+    }
+
+    // Handle primitives explicitly so the final fallback never receives an object.
+    const valueType = typeof value;
+    if (valueType === 'string') {
+        return value as string;
+    }
+    if (valueType === 'number' || valueType === 'boolean' || valueType === 'bigint' || valueType === 'symbol') {
+        const primitive = value as number | boolean | bigint | symbol;
+        return String(primitive);
+    }
+
+    if (valueType === 'function') {
```

Comment:
Can this be handled in the same block as `number`, `boolean`, etc...?

The [rule says](https://typescript-eslint.io/rules/no-base-to-string/):

> Note that Function provides its own .toString() and toLocaleString() that return the function's code. Functions are not flagged by this rule.